src/ir/opt/mem/lin.rs

```diff
@@ -69,12 +69,15 @@ impl RewritePass for Linearizer {
                         let idx_usize = extras::as_uint_constant(idx).unwrap().to_usize().unwrap();
                         Some(term![Op::Update(idx_usize); tup.clone(), val.clone()])
                     } else {
-                        let mut updates =
-                            (0..size).map(|idx| term![Op::Update(idx); tup.clone(), val.clone()]);
-                        let first = updates.next().unwrap();
-                        Some(key_sort.elems_iter().take(size).skip(1).zip(updates).fold(first, |acc, (idx_c, update)| {
-                        term![Op::Ite; term![Op::Eq; idx.clone(), idx_c], update, acc]
-                    }))
+                        let arr_idxs = (0..size).into_iter();
```

Comment:
`into_iter` is not needed here, because `0..size` is a `Range`, which [already implements `Iterator`](https://dev-doc.rust-lang.org/nightly/std/ops/struct.Range.html#impl-Iterator-for-Range%3CA%3E). I'd just inline `0..size` into `arr_idxs`'s use-site.